README.md

```diff
@@ -1,64 +1,53 @@
-One from the EndpointSlices when the host services are manually produced using the 'customEndpointSlices.json' file.
-The other matrix is generated by running 'ss' on the nodes.
-The test is expected to fail. You can find the output of the 'ss' command for each node and protocol,
-as well as the raw communication matrices in the 'e2etest/artifacts' directory, and the diff will be printed as part of the test output.
-
-### Communication Matrix Creation Guide
-
-The Communication Matrix is a structured list of Communication Details,  
-with each `ComDetails` entry representing a port. The fields for each entry  
-include `Direction` (currently "ingress" only), `Protocol` ("TCP" or "UDP"),  
-`Port` (number), `NodeRole` ("master" or "worker"), `ServiceName`,  
-and `Required` (false if optional).
-
-Struct Definitions:
-
-```
-type ComMatrix struct {
-	Matrix []ComDetails
-}
-
-type ComDetails struct {
-	Direction   string `json:"direction"`
-	Protocol    string `json:"protocol"`
-	Port        string `json:"port"`
-	NodeRole    string `json:"nodeRole"`
-	ServiceName string `json:"serviceName"`
-	Required    bool   `json:"required"`
-}
-```
-
-#### Usage of EndpointSlice Resource
+### Usage of the EndpointSlice Resource
 
 This library leverages the EndpointSlice resource to identify the ports the  
 cluster uses for ingress traffic. Relevant EndpointSlices include those  
 referencing host-networked pods, Node Port services, LoadBalancer services,  
 or any custom EndpointSlice labeled with `"ingress":""`.
 
-Explore the example in `/examples/query_endpointslices/main.go`.
+### Creating Custom ComDetails with ss Command
 
-#### Creating Custom ComDetails with ss Command
-
-To encompass all ports Kubernetes nodes are listening to, querying existing  
-EndpointSlices may be insufficient. Not all services, like the SSH service,  
-are represented. The `ss` command, a Linux utility, lists listening ports on  
+The `ss` command, a Linux utility, lists listening ports on  
 the host with `ss -anplt` for TCP or `ss -anplu` for UDP.
 
-The `ss` package provides the `ToComDetails` function, converting `ss` command  
-output into a corresponding ComDetails list. Use the `ToEndpointSlice` method  
-to create an EndpointSlice object from this list.
+The `ss` package provides the `CreateComDetailsFromNode` function that runs
+the `ss` command on each node, and converts the output into a corresponding ComDetails list.  
+
+### Communication Matrix Creation Guide
+
+Use the `generate` Makefile target to create the matrix.
+The following environment variables are used to configure:
+```
+FORMAT (csv/json/yaml)
+CLUSTER_ENV (baremetal/aws)
+DEST_DIR (path to the directory containing the artifacts)
+DEPLOYMENT (mno/sno)
+```
 
-As a convention, EndpointSlices referencing non-critical services are labeled with `"optional": ""`.
+The generated artifcats are:
+```
+communication-matrix.csv - The generated communication matrix.
```

Comment:
Should we avoid `.csv` since we allow to generate in several formats?

Reply:
yep, make sense. fixed